pkg/controller/storagecluster/update.go

```diff
@@ -117,6 +117,11 @@ func (c *Controller) rollingUpdate(cluster *corev1.StorageCluster, hash string,
 			nodesMarkedUnschedulable[node.Name] = true
 		}
 	}
+	// Non disruptive upgrade of nodes only if update strategy is rolling and if disruption is not false
+	if cluster.Spec.UpdateStrategy.RollingUpdate == nil || cluster.Spec.UpdateStrategy.RollingUpdate.Disruption.Allow == nil || !*cluster.Spec.UpdateStrategy.RollingUpdate.Disruption.Allow {
```

Comment:
I did try to do that, but it gave me an error - `mismatched types "github.com/libopenstorage/operator/pkg/apis/core/v1".Disruption and untyped nil` . Since it is of type "disruption" i cannot compare it to nil directly.

Reply:
can you check if Disruption is a pointer fields in RollingUpdateStorageCluster? If it is not, you will not be able to compare here. Let's change it to pointer if it is not and do correction here.

Reply:
Yes, let's change `Disruption` to pointer as well and then add that nil comparison here.